torchtext/models/roberta/modules.py

```diff
@@ -209,18 +213,27 @@ def __init__(
         self.final_layer_norm = nn.LayerNorm(embedding_dim)
         self.normalize_before = normalize_before
 
-    def forward(self, input, key_padding_mask):
+    def forward(self, input: torch.Tensor, key_padding_mask: torch.Tensor, attn_mask: Optional[torch.Tensor] = None):
+        if attn_mask is not None:
+            torch._assert(attn_mask.dtype == torch.bool, "Expected attn_mask dtype as `torch.bool` but got {}".format(attn_mask.dtype))
+            torch._assert(attn_mask.dim() == 2, "Expected attn_mask of dim 2 but got {}".format(attn_mask.dim()))
+            attn_mask = attn_mask.masked_fill(
+                attn_mask.to(torch.bool),
+                -1e8 if input.dtype == torch.float32 else -1e4
+            )
```

Comment:
Any particular reason for performing the masked fill in the layer forward as opposed to the encoder or self-attention forwards? (I am not advocating for one in particular over the others, mostly I am just curious)

Reply:
It's a good question. 

Regarding Encoder v/s Encoder layer: I think Encoder is client of Encoder layer. We could potentially user Encoder layer in other places too. So this choice is mainly about code re-usabilty.

Regarding Encoder layer v/s Multi head self attention: I think the choice here is less obvious. I think both are viable candidates to do this conversion. So based on your question, I actually made the conversion on both the modules and added support both for floating point and boolean mask. Let me know if this look better now.